sst/test/include/framework/expression.hh

```diff
@@ -0,0 +1,255 @@
+  uint64_t exprLineNum_;
+  ExprEval() : result_(0), is_binary_(0), exprString_(""), exprLineNum_(0){};
+  ExprEval(bool result, bool is_binary, std::string exprString,
+           uint64_t exprLineNum)
+      : result_(result),
+        is_binary_(is_binary),
+        exprString_(exprString),
+        exprLineNum_(exprLineNum){};
+  ExprEval(bool result, bool is_binary)
+      : result_(result),
+        is_binary_(is_binary),
+        exprString_(""),
+        exprLineNum_(0){};
+};
+
+/**
+ * This class represent an expression that can be evaluated to boolean a result.
+ */
+class BaseExpr {
+ public:
+  /**
+   * This is a virtual method which converts a Unary expression represented by a
+   * class extending BaseExpr into ExprEval.
+   */
+  virtual auto makeUnaryExprEval() -> ExprEval {
+    return ExprEval{false, false, "", 0};
+  }
+};
+
+/**
+ * This class represent the LHS of any expression which can be evaluated to a
+ * boolean result. For Unary expression the value evaluated is still stored
+ * as Lhs, however during evaluation the makeUnaryExprEval method uses the
+ * static_cast method for evaluation.
+ */
+template <typename Lhs, typename = IsFundamentalType<Lhs>>
+class LhsExpr : public BaseExpr {
+ private:
+  /** The left hand side of an expression. */
+  Lhs lhs_;
+
+ public:
+  explicit LhsExpr(Lhs lhs) : lhs_(lhs){};
+
+  /**
+   * This operator overload enables the execution of the 'greater than'
+   * operation on the rhs value without having to define its template type. The
+   * templated type is filtered by a SFINAE type trait which only allows
+   * template argument to be a fundamental type.
+   */
+  template <typename Rhs, typename = IsFundamentalType<Rhs>>
+  auto operator>(Rhs const& rhs) -> ExprEval {
+    return ExprEval{static_cast<bool>(lhs_ > rhs), true};
+  };
+
+  /**
+   * This operator overload enables the execution of the 'less than' operation
+   * on the rhs value without having to define its template type. The templated
+   * type is filtered by a SFINAE type trait which only allows the template
+   * argument to be a fundamental type.
+   */
+  template <typename Rhs, typename = IsFundamentalType<Rhs>>
+  auto operator<(Rhs const& rhs) -> ExprEval {
+    return ExprEval{static_cast<bool>(lhs_ < rhs), true};
+  };
+
+  /**
+   * This operator overload enables the execution of the 'greater than or
+   * equal' operation on the rhs value without having to define its template
+   * type. The templated type is filtered by a SFINAE type trait which only
+   * allows the template argument to be a fundamental type.
+   */
+  template <typename Rhs, typename = IsFundamentalType<Rhs>>
+  auto operator>=(Rhs const& rhs) -> ExprEval {
+    return ExprEval{static_cast<bool>(lhs_ >= rhs), true};
+  };
+
+  /**
+   * This operator overload enables the execution of the 'less than or equal'
+   * operation on the rhs value without having to define its template type. The
```

Comment:
Could you go through the file (/ project) and capitalise LHS and RHS please? Stylictic overkill I know...

Reply:
Done.